core/src/main/java/org/infinispan/registry/ScopedKey.java

```diff
@@ -46,6 +47,19 @@ public int hashCode() {
       return result;
    }
 
+   @Override
+   public String toString() {
+      Object oScope = scope;
+      Object oKey = key;
+      if (oScope instanceof MarshalledValue) {
```

Comment:
It is just because `LockManagerImpl` does so.  Originally, this request has been raised in a context of Query.  So `ScopedKey<String, Class>` used in query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java is likely the target case and `MarshalledValue` is not necessary actually.
Do you think the unmarshalling is not necessary only for scope or both of scope and key?